packages/type-safe-api/src/construct/integrations/s3.ts

```diff
@@ -40,6 +40,20 @@ export interface S3IntegrationProps {
    * @default - a combination of IntegrationResponseSets.defaultPassthrough() and IntegrationResponseSets.s3JsonErrorMessage()
    */
   readonly integrationResponseSet?: IntegrationResponseSet;
+
+  /**
+   * Specifies additional query string request parameters to be passed to the integration request.
+   * These parameters will be mapped to path parameters in order to allow for dynamic S3 bucket paths.
+   * @default - no additional query string request parameters
+   */
+  readonly queryStringRequestParameters?: string[];
+
+  /**
+   * Specifies additional header request parameters to be passed to the integration request.
+   * These parameters will be mapped to path parameters in order to allow for dynamic S3 bucket paths.
+   *  @default - no additional header request parameters
+   */
+  readonly headerRequestParameters?: string[];
```

Comment:
Nit: let's remove `These parameters will be mapped to path parameters in order to allow for dynamic S3 bucket paths.` :)